editoast/src/schema/rolling_stock.rs

```diff
@@ -0,0 +1,168 @@
+use diesel::result::Error as DieselError;
+use editoast_derive::EditoastError;
+use serde::{Deserialize, Serialize};
+use serde_json::Value as JsonValue;
+use thiserror::Error;
+
+#[derive(Debug, Insertable, Deserialize)]
+#[diesel(table_name = osrd_infra_rollingstock)]
+pub struct RollingStockForm {
+    pub name: String,
+    pub version: String,
+    pub effort_curves: JsonValue,
+    pub base_power_class: String,
+    pub length: f64,
+    pub max_speed: f64,
+    pub startup_time: f64,
+    pub startup_acceleration: f64,
+    pub comfort_acceleration: f64,
+    pub gamma: JsonValue,
+    pub inertia_coefficient: f64,
+    pub features: Vec<String>,
+    pub mass: f64,
+    pub rolling_resistance: JsonValue,
+    pub loading_gauge: String,
+    pub metadata: JsonValue,
+    pub power_restrictions: Option<JsonValue>,
+}
+
+#[derive(Debug, PartialEq, Queryable, Identifiable, Serialize)]
+#[diesel(table_name = osrd_infra_rollingstock)]
+pub struct RollingStock {
+    pub id: i64,
+    name: String,
+    version: String,
+    effort_curves: JsonValue,
+    base_power_class: String,
+    length: f64,
+    max_speed: f64,
+    startup_time: f64,
+    startup_acceleration: f64,
+    comfort_acceleration: f64,
+    gamma: JsonValue,
+    inertia_coefficient: f64,
+    features: Vec<String>,
+    mass: f64,
+    rolling_resistance: JsonValue,
+    loading_gauge: String,
+    metadata: JsonValue,
+    power_restrictions: Option<JsonValue>,
+}
+
+impl RollingStock {
+    pub async fn retrieve(db_pool: Data<DbPool>, rolling_stock_id: i64) -> Result<RollingStock> {
+        block::<_, Result<RollingStock>>(move || {
+            let mut conn = db_pool.get().expect("Failed to get DB connection");
+            match dsl::osrd_infra_rollingstock
+                .find(rolling_stock_id)
+                .first(&mut conn)
+            {
+                Ok(rolling_stock) => Ok(rolling_stock),
+                Err(DieselError::NotFound) => {
+                    Err(RollingStockError::NotFound { rolling_stock_id }.into())
+                }
+                Err(e) => Err(e.into()),
+            }
+        })
+        .await
+        .unwrap()
+    }
+
+    pub async fn create(
+        db_pool: Data<DbPool>,
+        rolling_stock_data: RollingStockForm,
+    ) -> Result<RollingStock> {
+        block::<_, Result<_>>(move || {
+            let mut conn = db_pool.get().expect("Failed to get DB connection");
+            match diesel::insert_into(dsl::osrd_infra_rollingstock)
+                .values(&rolling_stock_data)
+                .get_result::<RollingStock>(&mut conn)
+            {
+                Ok(rolling_stock) => Ok(rolling_stock),
+                Err(e) => Err(e.into()),
+            }
+        })
+        .await
+        .unwrap()
+    }
+
+    pub async fn delete(db_pool: Data<DbPool>, rolling_stock_id: i64) -> Result<()> {
+        block::<_, Result<_>>(move || {
+            use crate::tables::osrd_infra_rollingstock::dsl::*;
```

Comment:
why not used `dsl::osrd_infra_rollingstock` like in `create()` ?

Reply:
Hum I tried to remove it, but it doesn't work without this line :/ So I would rather keep it this way